### ISSUE TITLE
Needs work for 3.2

### ISSUE THREAD

**Issue by andrewvt** [unclecheese/silverstripe-event-calendar#85]
Calendar.php (line 438)
     return $events->getRange(0,$limit);
the method getRange() doesn't seem to exist anymore.

There are also some other issues affecting the CMS in general.  Could use a thorough 3.2 compatibility pass.

**Comment by 3Dgoo** [comment]
Yeah, there's a bit to fix. I started work on fixing some CMS issues but haven't got it to a 100% working stage yet:
https://github.com/3Dgoo/silverstripe-event-calendar/commit/ca3898fdf1bc0bdfc8036debe80591476ca7c25e

**Comment by unclecheese** [comment]
For the most part, there shouldn't be any API changes. `getRange()` was deprecated a while back, should be replaced with `limit()`. All those UI regressions you're seeing are due to the changes to form field IDs in 3.2, which were considered non-api changes. Hmph.

@3Dgoo your changes look good. Fire off a PR when you're ready and I'll merge. I haven't so much as looked at this module in over a year, so I appreciate the effort.

**Comment by 3Dgoo** [comment]
Happy to help. It's almost ready for a PR. @RuthAdele and I just need to do some final testing to check it works correctly in both 3.2 and 3.1.

**Comment by quamsta** [comment]
Thank you @3Dgoo, that will be a lifesaver. This has been preventing us from upgrading beyond 3.1 on numerous sites for months.

**Comment by 3Dgoo** [comment]
I have finally submitted a pull request #99 to fix this module for SilverStripe 3.2 and 3.3. From my testing it now works on versions 3.1, 3.2 and 3.3. Once this is merged in I'll keep an eye out for any issues that are lodged on github.

**Comment by quamsta** [comment]
# @3Dgoo is my hero!!! <3 <3
